crates/bevy_render/src/batching/gpu_preprocessing.rs

```diff
@@ -226,8 +226,14 @@ impl FromWorld for GpuPreprocessingSupport {
         let device = world.resource::<RenderDevice>();
 
         if device.limits().max_compute_workgroup_size_x == 0 ||
-            // filter lower end / older devices on Android as they crash when using GPU preprocessing
-            (cfg!(target_os = "android") && adapter.get_info().name.starts_with("Adreno (TM) 6"))
+            // filter some Qualcomm devices on Android as they crash when using GPU preprocessing
+            (cfg!(target_os = "android") && {
+                let name = adapter.get_info().name;
+                // filter out Adreno 730 and earlier GPUs (except 720, it's newer than 730)
+                name.strip_prefix("Adreno (TM) ").is_some_and(|version|
+                    version != "720" && version.parse::<u16>().unwrap() <= 730
```

Comment:
Unwrap here is bad if adreno includes letters in the version number later or something. `unwrap_or` or another `is_some_and` would be better.

Reply:
I think we can discard those non-numeric result anyways, since it will always newer than 730, so I changed unwrap to is_ok_and